scripts/neutron-band-structure.py

```diff
@@ -0,0 +1,209 @@
+    in the full set of labels from Seekpath.
+
+    """
+    # Find break points between continuous spectra: wherever there are two
+    # adjacent labels
+    special_point_bools = np.fromiter(
+        map(bool, bandpath["explicit_kpoints_labels"]), dtype=bool)
+    special_point_indices = special_point_bools.nonzero()[0]
+
+    # [T F F T T F T] -> [F F T T F T] AND [T F F T T F] = [F F F T F F] -> 3,
+    break_points = (np.logical_and(special_point_bools[:-1],
+                                   special_point_bools[1:])
+                    .nonzero()[0].tolist())
+
+    return break_points, special_point_indices
+
+
+def _get_tick_labels(region: Tuple[int, int],
+                     bandpath: dict,
+                     special_point_indices) -> List[Tuple[int, str]]:
+    start, end = region
+    label_indices = special_point_indices[
+        np.logical_and(special_point_indices >= start,
+                       special_point_indices < end)]
+
+    labels = [bandpath["explicit_kpoints_labels"][i]
+              for i in label_indices]
+    for i, label in enumerate(labels):
+        if label == 'GAMMA':
+            labels[i] = r'$\Gamma$'
+
+    label_indices_shifted = [i - start for i in label_indices]
+    return list(zip(label_indices_shifted, labels))
+
+
+def force_constants_from_file(filename):
+    path = pathlib.Path(filename)
+    if path.suffix == '.yaml':
+        force_constants = euphonic.ForceConstants.from_phonopy(
+            path=path.parent, summary_name=path.name)
+    elif path.suffix in ('.castep_bin', '.check'):
+        force_constants = euphonic.ForceConstants.from_castep(filename)
+    elif path.suffix == '.json':
+        force_constants = euphonic.ForceConstants.from_json_file(filename)
+    else:
+        raise ValueError("File not recognised. Should have extension "
+                         ".yaml (phonopy), .castep_bin or .check "
+                         "(castep) or .json (JSON from Euphonic).")
+
+    return force_constants
+
+def main():
+    args = get_parser().parse_args()
+    filename = args.file
+    summary_name = os.path.basename(filename)
+    path = os.path.dirname(filename)
+
+    print(f"Reading force constants from {filename}")
+    force_constants = force_constants_from_file(filename)
+
+    print(f"Getting band path")
+    structure = get_seekpath_structure(force_constants.crystal)
+    bandpath = seekpath.get_explicit_k_path(structure,
+                                            reference_distance=args.q_distance)
+
+    break_points, special_point_indices = _get_break_points(bandpath)
+    if break_points:
+        print(f"Found {len(break_points)} regions in q-point path")
+
+    regions = []
+    start_point = 0
+    for break_point in break_points:
+        regions.append((start_point, break_point + 1))
+        start_point = break_point + 1
+    regions.append((start_point, len(bandpath["explicit_kpoints_rel"])))
+
+    print("Computing phonon modes")
```

Comment:
It might be nice to tell the user how many q-points are being computed so they have a vague idea of how long it will take.

Reply:
This is straightforward in the current implementation, but could get more complex as further sampling schemes are added. I guess each mode of operation can be responsible for the message here. It would be cute to have a little progress bar through mod(q) (but add a dependency?)

Reply:
I guess for now we can just add the simple q-point calculation message and deal with it when things get more complicated as there aren't any compatibility issues with just changing how a message is printed.

I did look into a progress bar at one point (https://github.com/tqdm/tqdm) and adding it would be trivial for the Python case, but when calling down to C there's no way to report the progress back unless you periodically pass back to Python and I haven't done the research into whether a nice progress bar would be worth a performance hit (if any).

Reply:
My understanding is that well-designed progress bars improve _perceived_ performance which can compensate for overhead. There are papers on this but I'm not having much luck accessing them :disappointed:

Reply:
You're right, that would need to be hooked in at a lower level. Have implemented a count for now, should at least help people understand why it freezes when they request an absurdly dense path.